src/dist/notifications.rs

```diff
@@ -150,9 +148,6 @@ impl<'a> Display for Notification<'a> {
                 write!(f, "latest update on {date}, no rust version")
             }
             DownloadingLegacyManifest => write!(f, "manifest not found. trying legacy manifest"),
-            ManifestChecksumFailedHack => {
```

Comment:
I realized that this notification variant is used only once, so it makes sense to just inline it.

Reply:
... in the long run it might even make sense to remove the whole `notification_handler` mechanism?

Reply:
Yeah... I think we should unwind a lot of this `Notification` infrastructure.